src/components/general/Tabs/Tabs.tsx

```diff
@@ -13,45 +13,64 @@ type TabContentType = {
 };
 
 const TabContent: React.FC<TabContentType> = ({ children, activeTabKey }) => {
-    const active = React.Children.toArray(children).find(child => child.props.tabKey === activeTabKey);
+    const active = React.Children.toArray(children).find(
+        child => child.props.tabKey === activeTabKey
+    );
     if (!active) {
         return null;
     }
 
     const clonedChildren = React.Children.map(active.props.children, child =>
         React.cloneElement(child)
     );
-    return <React.Fragment>{clonedChildren}</React.Fragment>;
+    return <div className={styles.tabContent}>{clonedChildren}</div>;
 };
 
 const TabPane: React.FC<TabsProps> = ({ children, onChange, activeTabKey }) => {
+    const tabsPaneRef = React.useRef<HTMLDivElement | null>(null);
+
+    const scrollToTab = (tabRef: HTMLElement) => {
+        if (!tabsPaneRef.current || !tabRef) {
+            return;
+        }
+        const pane = tabsPaneRef.current;
+        const tab = tabRef;
+
+        if (pane.scrollWidth === pane.offsetWidth) {
+            return;
+        }
+        pane.scrollTo(tab.offsetLeft - pane.offsetWidth / 2 + tab.offsetWidth / 2, 0);
+    };
+
     const clonedChildren = React.Children.map(children, child => {
         const { title, tabKey } = child.props;
         if (!title || !tabKey) {
             return null;
         }
         return React.cloneElement(child, {
             onChange: () => onChange(tabKey),
+            onCurrent: scrollToTab,
```

Comment:
Couldn't this be solved with useEffect(scrollToTab, [activeTabKey]) ?

Reply:
But how would you then get the ref to the active tab?

Reply:
Set the ref here in the TabPane component and pass it along to the active child? 🤔

Reply:
Can't figure out how to pass the ref to the child, React.clone element won't let me pass refs as it is a function component. Do you have any suggestion on how this can be done? Could React.forwardRef() solve this in any way?

Reply:
forwardRef should allow you to accept refs, but I thought you should be able pass refs no matter what 🤔 Give it a try

Reply:
Didn't manage to solve it with forward ref.Instead  passed the tab ref to the parent as a part of the onChange function. Could then use useEffect(scrollToTab, [activeTabKey]) to scroll to the active tab